pg_replicate/src/pipeline/batching/data_pipeline.rs

```diff
@@ -54,6 +55,10 @@ impl<Src: Source, Snk: BatchSink> BatchDataPipeline<Src, Snk> {
         copied_tables: &HashSet<TableId>,
     ) -> Result<(), PipelineError<Src::Error, Snk::Error>> {
         let start = Instant::now();
+        self.source
```

Comment:
This is the fix for the transaction, by starting an explicit transaction here instead of relying on the constructor of PostgresSource to open a transaction.